### ISSUE TITLE
Idea: Web based(Server-less Javascript) or GUI Application for generating yml configuration.

### ISSUE THREAD

**Issue by antony-jr** [AppImageCrafters/AppImageCraft#1]
We have seen so many application authors wanting to integrate AppImages into their release system but they fail because they have very little knowledge about AppImages or think it as a chore. So I propose that we build some kind of server-less Javascript or GUI Application to generate the yml configuration file with respect to their Continuous Integration platform. **For starters we can aim at travis-ci**.

I can help to get this idea started, But I really don't know if its a good idea. Any suggestions?

**Comment by antony-jr** [comment]
Or we could also create a desktop applicatoin(GUI) to automatically modify the travis-ci files and also generate the **AppImageCraft configuration file**.

**Comment by azubieta** [comment]
I was thinking on create some sort of wizard like the one on Snapcraft with the common recipes. I guess that it can be extended with extra options to ease tweaking to the new-comers.

**Comment by antony-jr** [comment]
> wizard like the one on Snapcraft with the common recipes. I guess that it can be extended with extra options to ease tweaking to the new-comers.

Yes, also adjusting to the build process by parsing the travis.yml which needs some sort of guessing.

**Comment by antony-jr** [comment]
I think what we are going at is very similar to **[OBS](https://docs.appimage.org/packaging-guide/obs.html)**. OBS also builds AppImages automatically based on recipes but it cannot go well with continuous integration. Also it requires extra knowledge and a lot of chore to get started.

**Comment by azubieta** [comment]
> we can use the travis-ci api to encrypt the github api key directly into the travis.yml and automatically make the AppImages release onto a specific release tag like continuous. We could also create a Github App to integrate all these very smoothly.

Sounds terrific!

**Comment by antony-jr** [comment]
What I have in mind is this,
Build the App using **Next.js**(and Obviously React.js),

Using Github **OAuth** or Github token with repo scope get access to the user's targeted repo's write access and read .travis.yml and parse it to create a new .travis.yml integrating the AppImageCraft. Make sure to make a backup to revert the commit if something goes terribly wrong. 
*Also have a manual wizard to create a customized build with a template.*

If everything goes well, integrating AppImageCraft should be easy as **integrating codacy**. And that is awesome.

**Comment by azubieta** [comment]
I can feel that you're about to do it! Just let's consolidate a bit the implementation and we move on making the tool.

**Comment by azubieta** [comment]
Also it would be great if it integrates with gitlab and gitlab-ci. I guess that we could make a tool like this part of the https://opencode.net ecosystem.

**Comment by star-buck** [comment]
since travis-ci is not opensource by any means, i suggest to build this for gitlab ci for starters instead.

**Comment by antony-jr** [comment]
> since travis-ci is not opensource by any means, i suggest to build this for gitlab ci for starters instead.

@star-buck Thanks for the insight. But I think Travis-CI is open source too (https://github.com/travis-ci). Most of their tools are open source(Correct me if I'm wrong). But adding support for gitlab ci won't hurt.

**Okay I will be working on integrating this in a non-invasive way and also make it easy for the end user.**, The first revision will support Gitlab-CI and Github(Travis-CI).

@azubieta Some name suggestions could help(I'm very bad at naming stuff), also if I were to publish the source at some point, should I create a new repo in AppImageCraft org or in my personal account?

**Comment by azubieta** [comment]
> @azubieta Some name suggestions could help(I'm very bad at naming stuff), also if I were to publish the source at some point, should I create a new repo in AppImageCraft org or in my personal account?

What about naming it "AppImageCraft-Web", or just AppImageCraft. In a future we could host it in a domain with that name. 

Publishing the repo as part of the organization will be better to onboard possible collaborators. So, yes, please create the repo here.

**Comment by antony-jr** [comment]
Can we still use *AppImageCraft* as the name???

**Comment by azubieta** [comment]
The AppImageCraft project was there before we started this one, also appimage-builder is not a bad name. So let's avoid pointless discussions and keep using appimage-builder.
